File: House_Robber/house-robber.cpp
// https://leetcode.com/problems/house-robber/



class Solution {
public:
    int rob(vector<int>& nums) {
        
        int n = nums.size();
        int prev2 = 0, prev = nums[0];
        int cur = 0;
        
        for(int i=1; i<n; i++) {
            int rob = nums[i];
            if(i > 1) {
                rob += prev2;
            }
            
            int not_rob = 0 + prev;
            cur = max(rob, not_rob);
            
            prev2 = prev;
            prev=cur;
        }
        
        return prev;
        
    }
};

File: House_Robber/house-robber.py
"""
https://leetcode.com/problems/house-robber/
"""

class Solution:
    def rob(self, nums: List[int]) -> int:
        
        n = len(nums)        
        prev2 , prev =  0, nums[0]
        for i in range(1, len(nums)):
            
            rob = nums[i]
            if i > 1:
                rob+= prev2
            not_rob = 0 + prev
            
            cur = max(rob, not_rob)
            prev2 = prev
            prev = cur
            
        return prev